Use tight_layout to avoid clipping of feature names

HHest opened the pull request:
Without this, matplotlib could clip the feature names.

![image](https://user-images.githubusercontent.com/3169669/50164275-83e1f980-02e2-11e9-8f42-2b51dac20594.png)

vs.

![image](https://user-images.githubusercontent.com/3169669/50164214-51d09780-02e2-11e9-9bc9-eb941d1c1c49.png)

HHest committed:
tight_layout to avoid clipping of feature names

HHest committed:
tight_layout to avoid clipping of feature names

slundberg commented:
Sounds good, though it looks like it is causing some test failures at the moment.

HHest commented:
Yes, unsure why at moment... Test runs on my computer; I am using the same version 3.0.2 of matplotlib. Will try to reproduce.

rushkv committed:
UPD: performance comparison

Commit:
updated pytorch class for pytorch 1.0.0

slundberg committed:
Merge pull request #360 from canorbal/master

UPD: performance comparison

Commit:
interim layer fix

slundberg committed:
Update LinearExplainer and add sentiment analysis example

slundberg committed:
Add IMDB example dataset

slundberg committed:
Fix LinearExplainer tests

slundberg committed:
Merge pull request #361 from vlaqtz/master

updated pytorch class for pytorch 1.0.0

slundberg committed:
Example LinearExplainer notebook

slundberg committed:
Update README.md

slundberg committed:
Add large background check for DeepExplainer

slundberg committed:
Fix windows issue with set_printoptions

slundberg committed:
Avoid round() to make windows happy

slundberg committed:
Clean up unused files.

slundberg committed:
Remove vscode config

slundberg committed:
Better plots NaN values in dependence_plots

Works around a bug in matplotlib when coloring with NaN values in a scatter plot https://github.com/matplotlib/matplotlib/issues/4354

gabrieltseng committed:
Add interim layer tests for pytorch explainers

gabrieltseng committed:
Remove Pytorch 1.0 warnings from integrated gradient explainer

gabrieltseng committed:
Merge remote-tracking branch 'upstream/master' into pytorch_improvements

slundberg committed:
Update benchmark tests

slundberg committed:
Fix handling of DecisionTreeClassifier to normalize to prob

slundberg committed:
Bump version after fix

slundberg committed:
Merge pull request #378 from gabrieltseng/pytorch_improvements

PyTorch Improvements

slundberg committed:
Added tests to classification benchmark

slundberg committed:
Added ind tree shap

slundberg committed:
Update consistency garuntee list

slundberg committed:
Another consistency list update

slundberg committed:
Enable multi-output indenpendent tree shap

slundberg committed:
bump version

slundberg committed:
bump benchmark version

slundberg committed:
Partially fix #384

The real solution is to not let loss explanations also break

Tambah committed:
Fix issue #385

removed redundant "len" function

slundberg committed:
Include zeros in benchmark area tests

slundberg committed:
Add independentlinear60 test dataset

slundberg committed:
fix benchmark plot colors

slundberg committed:
Fix gbm output for benchmark

slundberg committed:
Bump benchmark version

slundberg committed:
Updated colors based on explicit Lch space choices

This gives a new better color set for multi-class bar charts, as well as a tweaked and improved global color scheme for plots.

slundberg committed:
Fix numerical precision mismatch with sklearn

slundberg committed:
Fix scikit-image dependency

slundberg committed:
Re-include colors needed by image_plot

slundberg committed:
Fix #393 and enable multiclass LinearExplainer

slundberg committed:
Fix XGBoost JSON bug by loading direct from memory

The JSON dump of an XGBoost model actually is not quite exact due to the conversion to base-10. I found this to cause samples to go down different paths in some cases. This fixes that by loading directly from memory.

slundberg committed:
Make py27 happy

slundberg committed:
Fix which class we explain for benchmark

slundberg committed:
Allow for a dynamic expected_value when explaining the loss

slundberg committed:
Rename variable to match count

slundberg committed:
Merge branch 'benchmark'

slundberg committed:
Fix #388

slundberg committed:
Fix indexing issue with benchmark

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Account for potentially small datasets in benchmark test

slundberg committed:
Merge pull request #386 from Tambah/fix_issue_385

Fix issue #385

slundberg committed:
Add support for rank based feature selection in KernelExplainer

This allows `l1_reg="rank(N)"` to be used to select the top N features as non-zero.

slundberg committed:
Depricate l1_reg="auto" in KernelExplainer

slundberg committed:
Fix typo in last KernelExplainer commit

slundberg committed:
Better load the objective of LightGBM

Fixed while working with the example in #392

slundberg committed:
Fix mac 10.14 builds

This fixes #399 #343 #369

slundberg committed:
Upgrade webpack-dev-server

slundberg committed:
Clean up version tag

slundberg committed:
Fix issues with expected_value and transformed outputs

slundberg committed:
Allow XGBoost node weights to be overridden

This addresses an issue in #90 #29 #352

slundberg committed:
Mirror the tree limit use of sklearn API XGB models

slundberg committed:
Fix tree_limit bug

HughChen committed:
Fix bug in tree_limit check

slundberg committed:
Merge pull request #402 from HughChen/patch-1

Fix bug in tree_limit check in shap_values()

pyrtsa committed:
Fix plot colors on Python 2.7 by using float division

slundberg committed:
Handle type size differences on windows.

Fixes #403

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix windows byte sizing issue during xgb load

slundberg committed:
Downgrade numpy api usage to support older numpy's

slundberg committed:
Fix #405

slundberg committed:
Merge pull request #404 from pyrtsa/future-division

Fix plot colors on Python 2.7 by using float division

slundberg committed:
Fix #407

Check for a string instead of an array

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix incorrect gray dots in summary plot #406

Gray dots should only be for nan feature values

slundberg committed:
Add save_html function

Related to #3

HughChen committed:
Add fix for l1reg

- Fix for floats
- Convert old "rank" to "num_features"

slundberg committed:
Merge pull request #410 from HughChen/patch-2

Add fix for l1reg

slundberg committed:
Fix issue with dependence_plot with no interaction

Fixes #411

slundberg committed:
Fix #424

Create extra random samples to prevent an overflow.

imatiach-msft committed:
fixed memory corruption error from TreeExplainer

slundberg committed:
Merge pull request #430 from imatiach-msft/ilmat/fix-tree-memory-err

fixed memory corruption error from TreeExplainer

jeremiahpslewis committed:
Fix deprecation warning in numpy v1.16

slundberg committed:
Merge pull request #434 from jlewis91/patch-1

Fix deprecation warning in numpy v1.16

Bacoknight committed:
Added support for skopt Random Forest and ExtraTrees Regressors. Added a small if statement to make the force graph a little nicer in the absence of feature values.

slundberg committed:
Merge pull request #436 from Bacoknight/master

Added support for skopt Random Forest and ExtraTrees Regressors.

jklaise committed:
Make kernel and sampling explainers have consistent nsamples defaults

vatsan committed:
Adding support for matplotlib forceplot with text rotation (tests, examples included)

slundberg committed:
Update README.md

slundberg committed:
Update README.md

slundberg committed:
Merge pull request #449 from vatsan/master

Adding support for matplotlib forceplot with text rotation

slundberg committed:
Merge pull request #448 from jklaise/nsamples_defaults

Make kernel and sampling explainers have consistent nsamples defaults

slundberg committed:
Bump release

slundberg committed:
Force appveyer to rebuild packages

slundberg committed:
Revamp the benchmark code

verdimrc committed:
Allow limited customizations to image_plot

Esp. useful for small input images like mnist.

slundberg committed:
Merge pull request #456 from verdimrc/verdimrc-patch-image_plot

Allow limited customizations to image_plot

slundberg committed:
Better docs for force_plot.

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

jorgecarleitao committed:
Added test verifying that linear shap gives the correct result.

jorgecarleitao committed:
Added test checking that DeepExplainer of a linear model (in tf keras) is correct.

jorgecarleitao committed:
Removed generic exception.

jorgecarleitao committed:
Added tensorflow installation to travis.

jorgecarleitao committed:
Fixed CI by installing tensorflow.

Also improved test skipping with nose.

jorgecarleitao committed:
Added Keras to travis and appveyor.

jorgecarleitao committed:
Added missing global_variables_initializer to test.

slundberg committed:
Merge pull request #468 from jorgecarleitao/feature/test_shap_values

Added test verifying that linear shap gives the correct result.

slundberg committed:
Fix unit test

slundberg committed:
Update test_deep.py

slundberg committed:
Merge pull request #473 from jorgecarleitao/fix/test_tensorflow

Fixed CI by installing tensorflow -- #472

slundberg committed:
Merge branch 'master' into feature/test_tf_keras_linear

slundberg committed:
Update test_deep.py

slundberg committed:
Update test_deep.py

slundberg committed:
Merge pull request #470 from jorgecarleitao/feature/test_tf_keras_linear

Added test checking that DeepExplainer of a linear model (in tf keras) is correct.

slundberg committed:
Fix #467

slundberg committed:
Update README.md

jpzhangvincent committed:
add new BalancedRandomForestClassifier model

slundberg committed:
Fix #484

slundberg committed:
Update README.md

slundberg committed:
Add brute force check

slundberg committed:
Address #427

Clarify the issues with position based features in text.

slundberg committed:
Update Keras LSTM for IMDB Sentiment Classification.html

slundberg committed:
Fix #490

This addresses some problems with the diagonal values of SHAP interaction values for internally computed models with multiple outputs. Also fixes an expected value problem where the base_offset was not included.

slundberg committed:
Tag new version with bug fixes.

slundberg committed:
Fix #489

Commit:
add BN to deep torch and fix a small Warning statement

slundberg committed:
Merge pull request #494 from JiechengZhao/pytorch_fix_bn

Add BN to deep torch and fix a small Warning statement

moritzaugustin committed:
fixed broken keras multi input model in gradient explainer

gabrieltseng committed:
Add leaky ReLU and other conv layers to pytorch deep explainer

slundberg committed:
Merge pull request #507 from gabrieltseng/pytorch_improvements

Add leaky ReLU and other conv layers to pytorch deep explainer

gabrieltseng committed:
Add fix to maxpool1d bug

moritzaugustin committed:
custom output rank

moritzaugustin committed:
seed argument for tf gradientexplainer shap_values

slundberg committed:
Merge pull request #504 from moritzaugustin/master

fixed broken keras multi input model in gradient explainer

moritzaugustin committed:
fixed seed in GradientExplainer constructor

slundberg committed:
Merge pull request #514 from moritzaugustin/master

fixed seed in GradientExplainer constructor

gabrieltseng committed:
Add warning about maxpool1d layer in pytorch deep explainer and set manual seed in test

gokceneraslan committed:
Set expected values in pytorch

Is this just missing or is there a special reason not to set self.expected_value in the constructor?

gokceneraslan committed:
Save expected_value for both single and multi output models

slundberg committed:
Merge pull request #517 from gokceneraslan/patch-1

Set expected values in pytorch

moritzaugustin committed:
fixing seed and added pytorch gradient seed

moritzaugustin committed:
improved seed documentation

slundberg committed:
Make errors like in #29 more clear

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix #518

Sklearn treats multiple regression different than multiple class clssifications. This fixes the code to correctly explain all outputs of a multiple regression problem.

imatiach-msft committed:
add XGBRanker and LGBMRanker to shap's TreeExplainer

slundberg committed:
Update README.md

slundberg committed:
Add bibtex files

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Address #535

slundberg committed:
Update README.md

slundberg committed:
Update gradient.py

doramir committed:
[FEAT] adding support for catBoost ranker

andriy-nikolov committed:
Issue #496: embedding lookup with tf.keras

Since tensorflow.keras uses the ResourceGather operation type for embedding lookup instead of GatherV2, registered gather as op handler for ResourceGather

slundberg committed:
Merge pull request #537 from doramir/adding-CatBoost

[FEAT] adding support for catBoost ranker

imatiach-msft committed:
updating based on comments

imatiach-msft committed:
added tests for ranking

slundberg committed:
Merge pull request #523 from moritzaugustin/master

Fixing seed in GradientExplainer

chua-quora committed:
make shap play nicer with sklearn<0.18

an alternative is to pin sklearn > 0.18

slundberg committed:
Merge pull request #552 from stillmatic/master

make shap play nicer with sklearn<0.18

imatiach-msft committed:
added validation

gabrieltseng committed:
Remove maxpool1d instability

slundberg committed:
Merge pull request #532 from imatiach-msft/ilmat/tree-ranker

add XGBRanker and LGBMRanker to shap's TreeExplainer

slundberg committed:
Merge pull request #516 from gabrieltseng/pytorch/maxpool

Fix maxpool1d in pytorch deep explainer

rcarneva committed:
dependence_plot uses color arg + optional cmap

slundberg committed:
Benchmark update

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix #527

slundberg committed:
Merge pull request #554 from rcarneva/master

dependence_plot uses color arg + optional cmap

slundberg committed:
Remove unneeded import

slundberg committed:
Drop Python 2 testing

Python 2 is EOL soon, so after version 0.28 I don't think we need to support it.

imatiach-msft committed:
fixed multirow sparse background case for KernelExplainer and added test

slundberg committed:
Fix #432

slundberg committed:
Merge pull request #557 from imatiach-msft/ilmat/fix-multirow-background-sparse

fixed multirow sparse background case for KernelExplainer and added test

slundberg committed:
Python 3.5 support for Windows

slundberg committed:
pylint fix

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Merge branch 'master' into master

slundberg committed:
Fix #558

slundberg committed:
Merge pull request #482 from jpzhangvincent/master

add new BalancedRandomForestClassifier model

slundberg committed:
Merge pull request #540 from andriy-nikolov/Issue-496-tf-embedding-lookup

Issue #496: embedding lookup with tf.keras

HHest committed:
Merge remote-tracking branch 'origin/patch-1' into patch-1

HHest commented:
I give up. Cannot figure out why the tests fail. Change isn't much, so not worth the effort.
